### DISCUSSION TITLE
Next.js Cache Poisoning vulnerability on latest react-email 3.0.1

### DISCUSSION THREAD

**Discussion by brandon-rilla** [resend/react-email#1734] [General]
Anyone else experiencing this security issue?

Looks like it's because react-email is using "Next.js between 13.5.1 and 14.2.9"

I might end up just removing the `react-email` dependency if they don't upgrade their dependecy to Next.js >14.2.9

<img width="912" alt="Screenshot 2024-10-14 at 7 48 55 PM" src="https://github.com/user-attachments/assets/ac9a9ea0-3318-4901-be31-11704556d0d5">

**Comment by pavelsushkov** [comment]
https://github.com/resend/react-email/security/advisories/GHSA-66rw-vq7j-wvq2

and as a temporary solution you can override this dependency in package.json

```json
"overrides": {
  "react-email": {
    "next": "^14.2.15"
  }
},
```

**Comment by brandon-rilla** [comment]
Thanks @pavelsushkov 

that was one of the things I tried but for some reason, npm didn't follow my overrides and still installed the old version in package-lock.json (I'm assuming this was some NPM bug)

The process to get this override working was pretty involved. I had to manually replace all incorrect versions of next inside `package-lock.json` with 14.2.15. Then I had to manually find, copy and paste the integrity SHAs directly from the nextjs registry.

Ideally react-email can update this security vulnerability themselves

**Comment by dinmukhamedm** [comment]
As an additional datapoint, I tried the ovverrides with `pnpm` and that didn't work out of the box.

I'm pretty sure that's worth opening an issue? @brandon-rilla do you want to open one from this discussion?

**Reply by dinmukhamedm** [reply]
Apparently fixed in https://github.com/resend/react-email/pull/1692 and staged for release in `react-email-3.0.2`
